crates/snark-acct/types/src/update.rs

```diff
@@ -241,6 +253,35 @@ impl SnarkAccountUpdate {
     }
 }
 
+/// SnarkAccountUpdate with mmr proofs of messages and ledger refs.
+///
+/// The sequencer is supposed to add the updated and valid mmr proofs to all message and ledger
+/// references in a `SnarkAccountUpdate` to construct `SnarkAccountUpdateWithMmrProofs`.
+#[derive(Clone, Debug)]
+pub struct SnarkAccountUpdateWithMmrProofs {
+    update: SnarkAccountUpdate,
+    message_proofs: Vec<MessageEntryProof>,
+    ledger_ref_proofs: LedgerRefProofs,
+}
```

Comment:
I've been saying for more than a year now how this naming convention is horrible and vague.  Created #1133 now.

For some resolution on a better way to model these types, see [this thread](https://github.com/alpenlabs/alpen/pull/1124#discussion_r2491584821) where we figured out a path forwards with this that's more systematic.  I'll be updating the specs to reflect this better organization.